lib/s3/models/BucketConfig.rb

```diff
@@ -172,6 +172,23 @@ def pretty_json
         }.reject { |k, v| v.nil? })
       end
 
+      # Public: deep sort an array of hashes or hash of arrays or array of arrays or hash of hashes or... you get the idea
+      #
+      #
+      # obj - The array or hash to deep sort
+      #
+      # Returns a deep sorted array of hashes or hash of arrays
+      def deep_sort_hash_array(obj)
```

Comment:
i'm thinking that the method should be modified so that we're not changing the object in place. I made a comment about it. the way the method is used, it doesn't expect it to change in place.